lib/inspec/dsl.rb

```diff
@@ -93,22 +93,29 @@ def self.load_spec_files_for_profile(bind_context, opts, &block)
     context = dep_entry.profile.runner_context
     # if we don't want all the rules, then just make 1 pass to get all rule_IDs
     # that we want to keep from the original
-    filter_included_controls(context, dep_entry.profile, &block) unless opts[:include_all]
+    filter_included_controls(context, dep_entry.profile, opts, &block) if !opts[:include_all] || !(opts[:conf]["profile"].include_tags_list.empty?)
+
     # interpret the block and skip/modify as required
     context.load(block) if block_given?
     bind_context.add_subcontext(context)
   end
 
-  def self.filter_included_controls(context, profile, &block)
+  def self.filter_included_controls(context, profile, opts, &block)
     mock = Inspec::Backend.create(Inspec::Config.mock)
     include_ctx = Inspec::ProfileContext.for_profile(profile, mock)
     include_ctx.load(block) if block_given?
+    include_ctx.control_eval_context.instance_variable_set(:@conf, opts[:conf])
```

Comment:
I had the same complaint on #5656 - we really should avoid instance_variable_Set on our own code whereever possible. we can make `conf` an attr_accessor on ControlEvalContext and set it properly.

Reply:
I understand now why you are suggesting to avoid using this. It goes against maintaining the class level encapsulations. Thanks for pointing this out. Have replaced this with `attr_accessor` usage.